### DISCUSSION TITLE
How to get the link of images in json

### DISCUSSION THREAD

**Discussion by abbasihamed** [fastapi/fastapi#6289] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI()

app.mount('/static', StaticFiles(directory='static'), name='static')

@app.get('/get/image')
async def get_image(db: Session = Depends(database.get_db)):
    img = db.query(store.Images).all()
    return img
```

### Description

Currently, my response is as follows:
```
[
  {
  "id":1,
  "image_url":"./static/image_name.png"
  }
]
```

But I expect images to be returned as urls in my response:
```
[
  {
  "id":1,
  "image_url":"http://Site_Address/static/image_name.png"
  }
]
```

### Operating System

Windows

### FastAPI Version

0.78.0

### Python Version

3.10.5

**Comment by JarroVGIT** [comment]
How do you store the url **string** in your db? If the full url is not in your db, how would any code automagically turn it into a full url? 

If they are stored like this, why not have the front end change it to whatever the FQDN is that the front end is running on?

**Comment by abbasihamed** [comment]
If you mean to add the url hard code, this is not correct
fastapi itself must have a method to get the url

**Comment by JarroVGIT** [comment]
Fastapi has no notion of where it is running from. You could distil it from the request object, I’d guess, but that would give you the requested URL at best. It could be enough for your use case though.

**Comment by JarroVGIT** [comment, marked as answer]
Something like this:

```python
from fastapi import Depends, FastAPI, Request

app = FastAPI()


def get_images_from_db():
    return [
        {"id": 1, "image_url": "./static/image_name1.png"},
        {"id": 2, "image_url": "./static/image_name2.png"},
    ]


@app.get("/get/image")
async def get_image(
    request: Request, img: list[dict[str, int | str]] = Depends(get_images_from_db)
):
    for img_obj in img:
        img_obj["image_url"] = str(request.base_url) + img_obj["image_url"][2:]
    return img


if __name__ == "__main__":
    import uvicorn

    uvicorn.run(app, host="0.0.0.0", port=80)
```

result with curl:
```bash
% curl localhost/get/image
[{"id":1,"image_url":"http://localhost/static/image_name1.png"},{"id":2,"image_url":"http://localhost/static/image_name2.png"}]%      
```

**Comment by abbasihamed** [comment]
Thanks
